globus_automate_client/cli/callbacks.py

```diff
@@ -64,6 +65,48 @@ def json_validator_callback(body: str) -> str:
     return body
 
 
+def yaml_validator_callback(body: str) -> str:
+    """
+    A user supplied body can be a properly formatted YAML string or a the name
+    of a file that contains valid YAML.  This validator ensures the user
+    supplies a valid YAML or valid filename containing valid YAML. Returns the
+    valid JSON string.
+    """
+    # Callbacks are run regardless of whether an option was explicitly set.
+    # Handle the scenario where the default value for an option is empty
+    if not body:
+        return body
+
+    # Reading from a file was indicated by prepending the filename with the @
+    # symbol -- for backwards compatability check if the symbol is present and
+    # remove it
+    if body.startswith("@"):
+        body = body[1:]
+
+    body_path = pathlib.Path(body)
+
+    if body_path.exists() and body_path.is_file():
+        with body_path.open() as f:
+            try:
+                yaml_body = yaml.safe_load(f)
+            except yaml.YAMLError as e:
+                raise typer.BadParameter(f"Invalid YAML: {e}")
```

Comment:
Would be curious what the error output from the exception looks like. If there's anything in the exception state that is helpful in tracking down the bad syntax (e.g. line/column numbers) AND if those are not presented nicely in the output from `{e}` it might be worth doing some custom formatting of the error message.

Reply:
Here is a screenshot of trying to parse invalid yaml:
![Invalid yaml parse](https://user-images.githubusercontent.com/76002510/103696459-012d4500-4f64-11eb-9af1-cbf987c3230f.png)